src/MiniCover/HitServices/MinimalHitService.cs

```diff
@@ -0,0 +1,146 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace MiniCover.HitServices
+{
+    public static class MinimalHitService
+    {
+        private static readonly object lockObject = new object();
+        private static readonly Dictionary<string, HitsTracer> files = new Dictionary<string, HitsTracer>();
+
+        public static void Init(string fileName)
+        {
+            lock (lockObject)
+            {
+
+                if (!files.ContainsKey(fileName))
+                {
+                    var fileInfo = new FileInfo(fileName);
+                    var tracer = files[fileName] = new HitsTracer();
+
+                    using (var streamReader = new StreamReader(fileInfo.Open(FileMode.OpenOrCreate, FileAccess.Read)))
```

Comment:
I think it's better to not read the hits file during execution and change save method to append or create file.